### ISSUE TITLE
GET from stage cannot handle remote path containing a sub-directory

### ISSUE THREAD

**Issue by dthacker82** [snowflakedb/snowflake-connector-net#688]
When attempting to retrieve a file from a sub-directory inside a stage, it fails with a generic "System.MethodNotImplemented" error. After decompiling the package and upon further debugging with breakpoints, it appears that the function that retrieves the file metadata pre-pends the sub-directory mapping to the local filename property.

`GET @stage/subdir/data_0_0_0.csv.gz file://c:\local\path`
will end up returning
`c:\local\path/subdir/data_0_0_0.csv.gz`
for the local filename, which is invalid.

Since the sub-directory doesn't matter to me in my use-case, I adjusted my code to unload to the `@stage` location, which it handles properly. Also, in this case, it is necessary to append a trailing `\` or `\\` depending on if/how you concatenate strings, whereas the SnowSQL client and ODBC driver do NOT need the trailing backslash in the local path when executing the GET command.

So, to summarize, the working syntax is
`GET @stage/data_0_0_0.csv.gz file://c:\local\path\`
which properly returns a local file of
`c:\local\path\data_0_0_0.csv.gz`

1. What version of .NET driver are you using?
2.0.25
   
2. What operating system and processor architecture are you using?
Windows 10 x64
   
3. What version of .NET framework are you using?
.NET 6.0 Standard

**Comment by sfc-gh-dszmolka** [comment]
hi and thank you for raising this issue, also for sharing the results of the debugging effort ! the issue is indeed reproducible, and while i could work around it with simply recreating the remote directory structure (i.e. in this example, creating `subdir` under `c:\local\path`), this is not the expected behaviour.  we'll take a look.

**Comment by sfc-gh-dszmolka** [comment]
#693

**Comment by sfc-gh-dszmolka** [comment]
PR is now merged and will be part of the next upcoming release, expected towards end of July

**Comment by sfc-gh-dszmolka** [comment]
PR is released as part of the latest Snowflake .NET driver version 2.1.0. Closing this issue.
